### ISSUE TITLE
accuracy of FedAvg is not equal to that of centralized training

### ISSUE THREAD

**Issue by NeutrinoLiu** [FedML-AI/FedML#113]
was trying the CI-script-fedavg.sh and ran the following: `assert that, for full batch and epochs=1, the accuracy of federated training(FedAvg) is equal to that of centralized training
`
> sh run_fedavg_standalone_pytorch.sh 0 1 1 -1 mnist ./../../../data/mnist lr hetero 10 1 0.03 sgd 0
> centralized_full_train_acc=$(cat wandb/latest-run/files/wandb-summary.json | python -c "import sys, json; print(json.load(sys.stdin)['Train/Acc'])")
> sh run_fedavg_standalone_pytorch.sh 0 1000 1000 -1 mnist ./../../../data/mnist lr hetero 10 1 0.03 sgd 0
> federated_full_train_acc=$(cat wandb/latest-run/files/wandb-summary.json | python -c "import sys, json; print(json.load(sys.stdin)['Train/Acc'])")
> assert_eq $(round $centralized_full_train_acc 3) $(round $federated_full_train_acc 3)
> cd ./../../../

the accuracy outcome is not equal: 0.358 != 0.822
any clue of why this happens?

**Comment by chaoyanghe** [comment]
How about using hyper-parameters suggested here?

https://github.com/FedML-AI/FedML/tree/master/benchmark

**Comment by NeutrinoLiu** [comment]
I believe your library works fine. It is simply because the two commands cannot fully emulate "full batch", considering the batch_size is [set to 128](https://github.com/FedML-AI/FedML/blob/3d1323d53d042d21d287b4a158247c14119e7d18/fedml_experiments/standalone/fedavg/main_fedavg.py#L102) when configured with full-batch training.

Accuracy of centralized training is much better when i manually modify the batch_size parameter to 61664 [instead of -1 ](https://github.com/FedML-AI/FedML/blob/3d1323d53d042d21d287b4a158247c14119e7d18/CI-script-fedavg.sh#L43)(where 61664 is full size of mnist). 

my suggestion is to remove [this assertion](https://github.com/FedML-AI/FedML/blob/3d1323d53d042d21d287b4a158247c14119e7d18/CI-script-fedavg.sh#L42) to avoid misleading.
